### DISCUSSION TITLE
[Feature Request] Add a chance to override the framework when vercel CLI can't read the framework

### DISCUSSION THREAD

**Discussion by itstheandre** [vercel/vercel#5198] [Ideas]
Hey there.

I have been playing around with vercel for a while now and I was wondering if this is at all possible.

When running vercel, to add some kind of flag for example `framework=nextjs` to let vercel know we are overriding it's framework "discovery feature". Dont know how else to put it.

I think this would be helpful if we have multiple deployment workflows and nextjs apps nested inside a monorepo. 

I am building stuff with people that want to do deployments via the CLI instead of doing it via git.

Is this possible or, would this be possible to do?

Because now the workflow I have is the following:

- run vercel, and vercel cli can't find the framework
- everything gets deployed, but what I see deployed is a file system. So I have to go the projects settings and change its framework from "Other" to Nextjs
- Redeploy with the new info

I wonder if this could be done with one simple command of `vercel --framework=next`

What do you think?

**Comment by paulogdm** [comment]
I am not sure about your use-case. Are you creating a lot of projects? If so, why?

**Reply by itstheandre** [reply]
I am not creating a lot of projects no, this goal is more , to be fully honest, a way to quickly override things from the CLI, rather than going to the dashboard. the same way we can change the build command, for example, we could maybe also change the framework.

does that make sense?
